### ISSUE TITLE
DropdownWidget: arrow is on the wrong position

### ISSUE THREAD

**Issue by cairabbit** [dotnetcore/BootstrapBlazor#196]
### Describe the bug
When showing a dropdown widget in the Header of Layout component, the arrow shows on the wrong position. The dropdown content also doesn't seem right.

### To Reproduce
```
<CascadingValue Value="this">
    <Layout ShowFooter="true">
        <Header>
        <nav class="navbar navbar-expand-lg navbar-light shadow">
            <div class="container d-flex justify-content-between align-items-center">
                <a class="navbar-brand text-success logo h1 align-self-center" href="/">
                    <img src="/img/favicon.png" alt="site logo"/>
                    <span>My Website</span>
                </a>
                <div class="align-self-center collapse navbar-collapse flex-fill d-lg-flex justify-content-lg-between" id="main_nav">
                    <Menu Items="@Menus" />
                </div>
                <SearchBox></SearchBox>
                <DropdownWidget class="px-3">
                    <DropdownWidgetItem Icon="fa fa-envelope-o" BadgeNumber="4">
                        <HeaderTemplate>
                            <span>您有 4 个未读消息</span>
                        </HeaderTemplate>
                        <BodyTemplate>
                            @for (var index = 0; index < 4; index++)
                            {
                                <a class="dropdown-item d-flex align-items-center" href="#" @onclick:preventDefault>
                                    <div style="width: 40px; height: 40px;">
                                        <Avatar Text="CC" IsText="true" IsCircle="true" Size="Size.Small" />
                                    </div>
                                    <div class="ms-2">
                                        <div class="d-flex position-relative">
                                            <h5>Message 1</h5>
                                            <small><i class="fa fa-clock-o"></i> @(4 + index) mins</small>
                                        </div>
                                        <div class="text-truncate">Why not buy a new awesome theme?</div>
                                    </div>
                                </a>
                            }
                        </BodyTemplate>
                        <FooterTemplate>
                            <a href="#" @onclick:preventDefault>查看所有消息</a>
                        </FooterTemplate>
                    </DropdownWidgetItem>
                </DropdownWidget>
            </div>
        </nav>
        </Header>
        <Main>
            @Body
        </Main>
        <Footer>
            
        </Footer>
    </Layout>
</CascadingValue>
```

### Exceptions (if any)
NA

### Further technical details
- BootstrapBlazor version 6.7.10
- Edge 102.0.1245.39 on windows 10

### Screen shots
![image](https://user-images.githubusercontent.com/1597596/173341426-17983be0-e002-499b-a468-4ee695476a6e.png)

**Comment by ArgoZhang** [comment]
@cairabbit hi, I'm not repro this issue. I think that maybe some css override the original css. Could you check it again? you can find them in the devtools on browser F12

**Comment by behzad62** [comment]
I have this problem too. Actually to reproduce this, you can just use the admin template and then move the widget component closer to the right edge (just put it after Logout component) and arrows will be misplaced.

**Comment by ArgoZhang** [comment]
@behzad62 thank you very much. i will repro and fixed it later

**Comment by ArgoZhang** [comment]
@cairabbit @behzad62 not repro yet. who can create a repo for this issue? let me check it

**Comment by ArgoZhang** [comment]
@behzad62 it works. thank you very much. I'm work on it

**Comment by ArgoZhang** [comment]
@cairabbit @behzad62 there is a new CSS named `dropdown-center` or `dropup-center` for this issue in `bootstrap` version `5.2.x`. I will upgrade this when it is released.

for now. you can use CSS overrides the `margin-left` property for owner project

**Comment by behzad62** [comment]
Good to know, waiting for the fix. Where should we override the `margin-left` and to what value?

**Comment by cairabbit** [comment]
@ArgoZhang sorry didn't response in time. I can see the width of the dropdown popup is a fix pixels, this is also not good to show some customized content. Is is possible to remove that width styling, but have the arrow at the right position? Thanks.

**Comment by ArgoZhang** [comment]
@cairabbit sure. there is a new CSS `dropdown-center` to ensure the dropdown menu aligns with the center of the input. but it does not release yet.

for now, fix the pixels to ensure the dropdown menu aligns with the center. I will remove the fixed pixels when the `bootstrap` release `5.2.x`

**Comment by ArgoZhang** [comment]
> Good to know, waiting for the fix. Where should we override the `margin-left` and to what value?

@behzad62 sorry. I really do not know what value is fit your widget. you can change the value using `Arrow up` and `Arrow down` in the dev tool. find the right value.
